### PR TITLE
feat(modules/home-manager): add support for configuring environment variables

### PR DESCRIPTION
**Pull request by marksisson**
Add support for configuring environment variables before calling `sops-install-secrets`. Introduce a new `environment` option which allows specifying environment variables. Modify systemd service and launchd agent to use the specified environment variables.

### CODE REVIEW

File: modules/home-manager/sops.nix [hunk 1]
```diff
@@ -174,6 +171,18 @@ in {
...
     };
 
+    environment = lib.mkOption {
+      type = lib.types.attrsOf (lib.types.either lib.types.str lib.types.path);
+      default = {};
+      description = ''
+        Environment variables to set before calling sops-install-secrets.
+
+        The values are placed in single quotes and not escaped any further to
+        allow usage of command substitutions for more flexibility. To properly quote
```
Comment (Mic92) [inline modules/home-manager/sops.nix:181]: Is this actually true? I don't think systemd will do command substitution.

Comment (Mic92) [reply modules/home-manager/sops.nix:181]: I don't now about launchd, but it also looks to me that it should be the same.

Comment (marksisson) [reply modules/home-manager/sops.nix:181]: That may not be true. I copied the description for the environment option from the NixOS module at modules/sops/default.nix.

I also have only tested the launchd code path. It does work on macOS. I do not have access to a NixOS linux system. Hopefully someone can test the systemd code path.

**Comment by Mic92** [comment]
@SebTM could you test linux?

**Comment by SebTM** [comment]
Hey, sorry for the delay - I can test it but would appreciate a test-case?

**Comment by Mic92** [comment]
@SebTM for now it's enough if you just make sure it doesn't break your existing usage.

**Comment by Mic92** [comment]
@mergify queue

**Comment by Mic92** [comment]
@mergify rebase

**Pull request merged**